src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3743,12 +3744,42 @@ ImmutableArray<PropertySymbol> addProperties(ImmutableArray<ParameterSymbol> rec
                     if (!memberSignatures.TryGetValue(targetProperty, out var existingMember))
                     {
                         existingMember = OverriddenOrHiddenMembersHelpers.FindFirstHiddenMemberIfAny(targetProperty, memberIsFromSomeCompilation: true);
-                        isInherited = true;
+                        if (existingMember is FieldSymbol)
+                        {
+                            // field from base type should not be considered before we look at fields from current type
+                            existingMember = null;
+                        }
+                        else
+                        {
+                            isInherited = true;
+                        }
+                    }
+
+                    if (existingMember is null)
+                    {
+                        var targetField = new SignatureOnlyFieldSymbol(param.Name,
+                                                                       this,
+                                                                       param.TypeWithAnnotations);
+
+                        if (!memberSignatures.TryGetValue(targetField, out existingMember))
+                        {
+                            existingMember = OverriddenOrHiddenMembersHelpers.FindFirstHiddenMemberIfAny(targetField, memberIsFromSomeCompilation: true);
+                            if (existingMember is not FieldSymbol)
+                            {
+                                existingMember = null;
+                            }
+                        }
                     }
+
                     if (existingMember is null)
                     {
                         addProperty(new SynthesizedRecordPropertySymbol(this, syntax, param, isOverride: false, diagnostics));
                     }
+                    else if (existingMember is FieldSymbol { IsStatic: false } field)
```

Comment:
>existingMember is FieldSymbol { IsStatic: false } field [](start = 29, length = 55)

Are we making sure the type matches? Where?

Reply:
That was missing. Thanks

---
In reply to: [612811257](https://github.com/dotnet/roslyn/pull/52480#discussion_r612811257) [](ancestors = 612811257)